From ca73c1973b055e089c7e44c83dc4eebfc2c9abc6 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Fri, 14 Feb 2025 14:45:08 +0100
Subject: [PATCH] dt-bindings: stm32: create st,stm32mp2-ca35ss-clk DT binding

Create st,stm32mp2-ca35ss-clk DT binding.

Change-Id: Ied66e943fba97a49a80b7a13820e219b88f86337
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/437298
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 .../clock/st,stm32mp2-ca35ss-clk.yaml         | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/clock/st,stm32mp2-ca35ss-clk.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/clock/st,stm32mp2-ca35ss-clk.yaml
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/clock/st,stm32mp2-ca35ss-clk.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STMicroelectronics CA35SS Clock Controller on STM32MP
+
+maintainers:
+  - Theo GOUREAU <theo.goureau@foss.st.com>
+
+description: |
+  This binding describes the clock controller provided by the secure monitor
+  (TF-A BL31) which supplies the clock for the Cortex-A35 subsystem (CA35SS).
+  The clock controller uses SMC calls for frequency scaling.
+
+properties:
+  compatible:
+    const: "st,stm32mp2-ca35ss-clk"
+
+  "#clock-cells":
+    const: 0
+
+required:
+  - compatible
+  - "#clock-cells"
+
+additionalProperties: false
+
+examples:
+  - |
+      ca35ss_clk: ca35ss-clk {
+        compatible = "st,stm32mp2-ca35ss-clk";
+        #clock-cells = <0>;
+      };
+...
